src/ate/test_programs/cp.cc

```diff
@@ -114,15 +114,15 @@ absl::StatusOr<std::string> ValidateFilePathInput(std::string path) {
       absl::StrCat("Unable to open file: \"", path, "\""));
 }
 
-std::string BytesToHexStr(const char *bytes, size_t len) {
-  std::stringstream ss;
-  ss << std::hex << std::uppercase << std::setfill('0');
-  for (size_t i = 0; i < len; ++i) {
-    ss << std::setw(2)
-       << static_cast<int>(static_cast<unsigned char>(bytes[i]));
+bool SetDiversificationString(uint8_t *diversifier, const std::string &str) {
+  if (str.size() > 48) {
+    return false;
   }
-  return ss.str();
+  memcpy(diversifier, str.c_str(), str.size());
```

Comment:
nit: You can use `str.data()` here because you're performing a memcpy with `size()` and you don't care about the `nul` terminator (which is what `c_str()` provides).

Reply:
Fixed.